Reorganize Enterprise Output Framework docs and add BigQuery Output (#935)
Graylog2/documentation: pages/integrations/output_framework/output_google_bigquery.rst

Create Service Account
^^^^^^^^^^^^^^^^^^^^^^

1) Log in to the `Google Cloud console <https://console.cloud.google.com>`_
2) Navigate to ``IAM & Admin`` in the left-hand menu
3) Select ``Service Accounts`` in the left-hand menu
4) Select ``+ CREATE SERVICE ACCOUNT`` in the top of the right-hand pane
5) Create the new service account

   a) Provide a name for the service account (i.e. "Graylog Data")
   b) Enter a description for the service account
   c) Click the ``CREATE`` button
   d) Select appropriate permissions for the new service account.  At a minimum, the service 
      account will need the ability to write to your BigQuery table
   e) Click the ''CONTINUE'' button
   f) If desired, grant other users access to the service account
   g) Click the ''DONE'' button to finish service account creation

Generate and Download Service Account Credentials
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1) Click on the newly created service account in your list of service accounts
2) In the ``Keys`` section, select ``Create new key`` from the ``ADD KEY`` drop-down menu
3) Select ``JSON`` as the key type
4) Click on the ``CREATE`` button
5) Save the generated JSON file

> 6) Copy the downloaded JSON credentials file to your Graylog host(s).  We strongly 

Reviewer: For here too, As we specify the "Credentials File Location", and that is only 1 path, we should make a note to put them in the same location on all nodes.

Author: Good call

   recommend that you take appropriate steps to protect the credentials file (e.g.
   assigning ownership of the file to the account which runs your Graylog server and 
   setting file permissions to 400).
   
Output Configuration
--------------------

The Google Cloud BigQuery Output supports all of the standard Enterprise Output Framework 
`configuration options <../output_framework.html#general-configuration>`__.

BigQuery Configuration
^^^^^^^^^^^^^^^^^^^^^^

- ``Project ID``
   - Google Cloud Project ID
- ``Dataset``
   - Output BigQuery Dataset
- ``Table``
   - Output BigQuery Table
- ``Excluded Fields``

>    - A comma-separated list of fields that will be used as filter which fields are sent to BigQuery

Reviewer 2: ```suggestion
   - A comma-separated list of fields that are filtered and is not sent to BigQuery
```

Author: Will fix.

- ``Credentials File Location``

>    - Path to the Service Account credentials file

Reviewer: Maybe expand a bit here:

Path to the Service Account credentials file, located on your Graylog Node(s).

Author: Will do
